File: catkin_ws/src/wave_judger/scripts/image_publisher.py
import cv2
import rospy
from sensor_msgs.msg import Image
import numpy as numpy
from cv_bridge import CvBridge, CvBridgeError
import sys

def PublishImages():
    rospy.init_node('image_publisher', anonymous=True)
    img_pub = rospy.Publisher('camera/image', Image, queue_size=1)
    rate = rospy.Rate(20)

    cap = cv2.VideoCapture(0)
    bridge = CvBridge()

    while not rospy.is_shutdown():
        ret, frame = cap.read()

        if ret == 0:
            break
        
        msg = bridge.cv2_to_imgmsg(frame, encoding='bgr8')
        img_pub.publish(msg)
        rate.sleep()

if __name__ == "__main__":
    try:
        PublishImages()
    except rospy.ROSInterruptException:
        pass

File: catkin_ws/src/wave_judger/src/image_publisher.cpp
#include <ros/ros.h>
#include <image_transport/image_transport.h>
#include <opencv2/opencv.hpp>
#include <cv_bridge/cv_bridge.h>

int main(int argc, char **argv){
    ros::init(argc, argv, "video_transp");
    ros::NodeHandle nh;
    image_transport::ImageTransport it(nh);
    
    //the second param refer to length of buffer
    image_transport::Publisher pub = it.advertise("camera/image", 1);
    // cv::Mat image = cv::imread(argv[1], CV_LOAD_IMAGE_COLOR);
    sensor_msgs::ImagePtr msg;
    ros::Rate loop_rate(10);

    cv::VideoCapture cap(0);
    cv::Mat frame;
    while (nh.ok()){
        cap >> frame;
        if (!frame.empty()){
            msg = cv_bridge::CvImage(std_msgs::Header(), "bgr8", frame).toImageMsg();
            pub.publish(msg);
        }
        ros::spinOnce();
        loop_rate.sleep();
    }           

    return 0;
}